habitat/plan.sh

```diff
@@ -0,0 +1,59 @@
+pkg_name=knife-ec-backup
+pkg_origin=chef
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_description="A knife plugin for backup/restore of Chef Server data."
+pkg_upstream_url="https://github.com/chef/knife-ec-backup"
+pkg_license=('Apache-2.0')
+pkg_bin_dirs=(bin)
+pkg_lib_dirs=(lib)
+pkg_svc_user=root
+pkg_svc_group=${pkg_svc_user}
+pkg_build_deps=(
+  core/gcc-libs
+  core/libffi
+  core/make
+  core/openssl
+)
+pkg_deps=(
+  core/bundler
+  core/coreutils
+  core/gcc
+  core/ruby
+)
+
+pkg_build_deps=(
+  core/git
+  core/make
+  core/postgresql-client
+)
+
+do_unpack() {
+  mkdir -p "${HAB_CACHE_SRC_PATH}/${pkg_dirname}"
+  cp -r ../* ${HAB_CACHE_SRC_PATH}/${pkg_dirname}/
+}
+
+
+do_build() {
+  pushd "${HAB_CACHE_SRC_PATH}/${pkg_dirname}"
+  bundle install --jobs 2 --retry 5 --path ./vendor/bundle --binstubs
+  popd
```

Comment:
Same for this block. I think we recommend the use of sub-shells:
```bash
  ( cd "${HAB_CACHE_SRC_PATH}/${pkg_dirname}" | exit_with "unable to enter cache directory" 1
    bundle install --jobs 2 --retry 5 --path ./vendor/bundle --binstubs
  )
```

Reply:
This pushd pattern is used extensively in the core plans: https://github.com/habitat-sh/core-plans/search?q=pushd&unscoped_q=pushd

Reply:
It is indeed but (as far as I know) we have been trying to move away from it since `pushd/popd` are not safe because you could end up with an altered `CWD` if something happens in between them, instead we would like to use subshells to isolate a particular subset of commands that needs to run on a specific path, this pattern will guarantee that at the end, you will have the original `CWD`, regardless how the execution happens. (just like your case)

**Caveat**: `pushd/popd` are easier to read.